app/internal/google_connect.py

```diff
@@ -0,0 +1,230 @@
+from datetime import datetime
+from fastapi import Depends
+
+from google.auth.transport.requests import Request as google_request
+from google.oauth2.credentials import Credentials
+from google_auth_oauthlib.flow import InstalledAppFlow
+from googleapiclient.discovery import build
+
+from app.database.models import Event, User, OAuthCredentials, UserEvent
+from app.dependencies import get_db, SessionLocal
+from app.config import CLIENT_SECRET_FILE
+from app.routers.event import create_event
+
+
+SCOPES = ['https://www.googleapis.com/auth/calendar']
+
+
+def get_credentials(user: User,
+                    session: SessionLocal = Depends(get_db)) -> Credentials:
+    credentials = get_credentials_from_db(user)
+
+    if credentials is not None:
+        credentials = refresh_token(credentials, session, user)
+    else:
+        credentials = get_credentials_from_consent_screen(
+            user=user, session=session)
+
+    return credentials
+
+
+def fetch_save_events(credentials: Credentials, user: User,
+                      session: SessionLocal = Depends(get_db)) -> None:
+    if credentials is not None:
+        events = get_current_year_events(credentials, user, session)
+        push_events_to_db(events, user, session)
+
+
+def clean_up_old_credentials_from_db(
+    session: SessionLocal = Depends(get_db)
+) -> None:
+    session.query(OAuthCredentials).filter_by(user_id=None).delete()
+    session.commit()
+
+
+def get_credentials_from_consent_screen(user: User,
+                                        session: SessionLocal = Depends(get_db)
+                                        ) -> Credentials:
+    credentials = None
+
+    if not is_client_secret_none():  # if there is no client_secrets.json
+        flow = InstalledAppFlow.from_client_secrets_file(
+            client_secrets_file=CLIENT_SECRET_FILE,
+            scopes=SCOPES
+        )
+
+        flow.run_local_server(prompt='consent')
+        credentials = flow.credentials
+
+        push_credentials_to_db(
+            credentials=credentials, user=user, session=session
+        )
+
+        clean_up_old_credentials_from_db(session=session)
+
+    return credentials
+
+
+def push_credentials_to_db(credentials: Credentials, user: User,
+                           session: SessionLocal = Depends(get_db)
+                           ) -> OAuthCredentials:
+
+    oauth_credentials = OAuthCredentials(
+        owner=user,
+        token=credentials.token,
+        refresh_token=credentials.refresh_token,
+        token_uri=credentials.token_uri,
+        client_id=credentials.client_id,
+        client_secret=credentials.client_secret,
+        expiry=credentials.expiry
+    )
+
+    session.add(oauth_credentials)
+    session.commit()
+    return credentials
+
+
+def get_active_user(session: SessionLocal = Depends(get_db)) -> User:
+    # TODO - get connected/current user
+    user = session.query(User).filter_by(id=1).first()
+    return user
```

Comment:
Use the login system - it has some pretty easy API :)

Reply:
Done.